### PR TITLE
Add app insights agent spring 1.x

### PR DESCRIPTION
**Pull request by timja**

### CODE REVIEW

File: lib/AI-Agent.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ApplicationInsightsAgent>
+    <AgentLogger>INFO</AgentLogger>
+    <Instrumentation>
+        <BuiltIn enabled="true">
+            <HTTP enabled="true"/>
+        </BuiltIn>
+        <Class name="uk.gov.hmcts.cmc.claimstore.repositories.DBCaseRepository">
```
Comment (mario-paniccia) [inline lib/AI-Agent.xml]: currently updating these to CCD repositories

File: lib/AI-Agent.xml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            <Method name="findCaseRoles"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.caseaccess.DefaultCaseRoleRepository">
+            <Method name="getCaseRoles"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.user.DefaultUserRepository">
+            <Method name="getUserDetails"/>
+            <Method name="getUserDefaultSettings"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.definition.UIDefinitionRepository">
```
Comment (mario-paniccia) [inline lib/AI-Agent.xml]: @timja some of our repositories, for example UIDefinitionRepository, do not interact with the database. They instead issue http calls to external services. Shall we include them as well or this file if purely for database layer Repositories?

File: lib/AI-Agent.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            <Method name="getBaseTypes"/>
+            <Method name="getUserRoleClassifications"/>
+            <Method name="getClassificationsForUserRoleList"/>
+            <Method name="getLatestVersion"/>
+            <Method name="getJurisdictions"/>
+        </Class>
+        <Class name="uk.gov.hmcts.ccd.data.casedetails.DefaultCaseDetailsRepository">
+            <Method name="set"/>
+            <Method name="findById"/>
+            <Method name="findByReference"/>
```
Comment (mario-paniccia) [inline lib/AI-Agent.xml]: @timja we have overloaded findByReference with different number of parameters. Will this line instrument them all?

File: lib/AI-Agent.xml (hunk 2 above)
Comment (timja) [reply lib/AI-Agent.xml]: no http should be automatically instrumented, but needs to be verified. I'll do that soon unless you beat me to it

File: lib/AI-Agent.xml (hunk 3 above)
Comment (timja) [reply lib/AI-Agent.xml]: I think so, you can pick and choose with something like this:
https://github.com/hmcts/cmc-claim-store/blob/master/lib/AI-Agent.xml#L30

useful if i.e. all of the overloads delete to one method and then you just put the method they all call there

**Comment by timja** [comment]
This is what it currently looks like:
image

It's not tracing through from the frontend properly, not sure what that 320ms CASE_DEFINITION is

Will look into it some more

https://portal.azure.com/#blade/AppInsightsExtension/DetailsV2Blade/DataModel/H4sIAAAAAAAAA5VTy27bMBD8FYOnFqBsSdTTCHpJ08CH9pC4PrTogaJWNluKUvhwEwT%252B9yxlO03b9FABAqTV7sxyZvRIxKCd4cJtwFg5aLIkCaFY7cdBg3arliwfySfeA35xsv8eObAOO259Y4WRo8Oh0ESajsUVz0UUF2kWZYxlUdV1VVS1gtddm7GyYjh3A3bwRsC1Gfz4C7MfyYGSYQTDz4jaK0UJ7E9bkDwtawRkUZbwLkoSqKOma5CqZiKtc9HhTU4Da96osLGBO4%252F7WqwjET7xPpCmcVJHMYviap1kS5Yts3xeltmXU9slagL3Lpy89ceFPlqyrIosDhdy6HYtJ0n%252BQCqWcTzHloAkDHAH%252F2xk%252BbwoJkppV1o6ydWxtePKAiVbwyXakVDiLbzndtcM3ExoN1xvz30omuBiB5NCbdOlRV2jJLwq0YKkipqU5RErGyiaMs2TOkM%252BAx0YAyaczz2MCMUo0UeLb4EbsdskwQ0n3STi9dV69nl1ye0kjBmUArPYggsVhLPTyAepHEZowjyJ7zUKNwvpAkvPTtCRb2Ej4aelwls39FfBL0vhXsAUJktbGFFg0ELiHN9zqXgjlXQPmB2vJjO7iW2FuUHGr4%252Fkx506Zulo8ezdrEXtw%252FubZ%252BHL3x16izAIDCqsrJ8DfgbAr3uufKjG5SJmi4BDZ2cIcjjQv2kvXqGt%252Fof24iVt9SrtN%252FxN9uifbOFFGJzxcDg8AZOsvxDPAwAA/ComponentId/%7B"Name"%3A"timj-test"%2C"SubscriptionId"%3A"bf308a5c-0624-4334-8ff8-8dca9fd43783"%2C"ResourceGroup"%3A"timj-temp"%7D

**Pull request merged**